### PR TITLE
[Dev] fetch total unread messages and set UI badges to count

### PR DESCRIPTION
**Pull request by l2succes**
Related: https://github.com/artsy/collector-experience/issues/812

We fetch the `totalUnreadCount` from metaphysics when the app becomes active and set the app and Inbox tab badge to that value.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Artsy/App/ARAppNotificationsDelegate.m [hunk 1]
```diff
@@ -193,12 +195,6 @@ - (void)applicationDidReceiveRemoteNotification:(NSDictionary *)userInfo inAppli
             NSUInteger count = [userInfo[@"aps"][@"badge"] unsignedLongValue];
             [[ARTopMenuViewController sharedController] setNotificationCount:count forControllerAtIndex:ARTopTabControllerIndexMessaging];
         }
-        // Set the badge count on the tab that the view controller belongs to.
-//        NSInteger tabIndex = [[ARTopMenuViewController sharedController] indexOfRootViewController:viewController];
-//        if (tabIndex != NSNotFound) {
-//            NSUInteger count = [userInfo[@"aps"][@"badge"] unsignedLongValue];
-//            [[ARTopMenuViewController sharedController] setNotificationCount:count forControllerAtIndex:tabIndex];
-//        }
```
Comment (alloy) [inline Artsy/App/ARAppNotificationsDelegate.m]: This needs to be re-instated so the badge count gets updated when a push notification comes in.

File: Artsy/Networking/API_Modules/ArtsyAPI+CurrentUserFunctions.m [hunk 2]
```diff
@@ -36,5 +35,22 @@ + (void)getCurrentUserLotStandingsForSale:(NSString *)saleID success:(void (^)(N
     [self getRequest:request parseIntoAnArrayOfClass:[LotStanding class] success:success failure:failure];
 }
 
++ (void)getCurrentUserTotalUnreadMessagesCount:(NSInteger)count success:(void (^)(NSInteger))success failure:(void (^)(NSError *error))failure
```
Comment (alloy) [inline Artsy/Networking/API_Modules/ArtsyAPI+CurrentUserFunctions.m]: What’s the `count` parameter for?

File: Artsy/Networking/conversations.graphql [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+    me {
+        conversations(first: %@) {
```
Comment (alloy) [inline Artsy/Networking/conversations.graphql]: Can’t this be `0` ? We really only need `totalUnreadCount`, no edges.

File: Artsy/App/ARSpotlight.m [hunk 4]
```diff
@@ -132,9 +132,11 @@ + (void)indexAllUsersFavorites;
...
     dispatch_block_t finalizeBlock = ^{
 #ifdef DEBUG
-        if (application.applicationState == UIApplicationStateBackground) {
-            NSLog(@"Remaining allowed background time by task finalizing: %f", application.backgroundTimeRemaining);
-        }
+        ar_dispatch_main_queue(^{
+            if (application.applicationState == UIApplicationStateBackground) {
+                NSLog(@"Remaining allowed background time by task finalizing: %f", application.backgroundTimeRemaining);
+            }
+        });
```
Comment (alloy) [inline Artsy/App/ARSpotlight.m]: Ooh, good call, I think we have a ticket for this. But actually this reminds me that we wanted to just disable spotlight indexing completely for now, I’ll make a PR for that before cutting the beta.

File: Artsy/App/ARAppNotificationsDelegate.m (hunk 1 above)
Comment (l2succes) [reply Artsy/App/ARAppNotificationsDelegate.m]: we're already updating the badge count here. https://github.com/artsy/eigen/pull/2507/files/322fdaf729f922f46a068f36618ce5e621e67ed1#diff-811684a49439e5089bba6275d1b55c79R196
Although it'll only set it for messaging push notifications (with a `/conversation/` url), should I just remove the code above and keep that block instead?

Comment (alloy) [reply Artsy/App/ARAppNotificationsDelegate.m]: Oooh, right, my bad. No this is good for now 👍

File: Artsy/Networking/API_Modules/ArtsyAPI+CurrentUserFunctions.m (hunk 2 above)
Comment (l2succes) [reply Artsy/Networking/API_Modules/ArtsyAPI+CurrentUserFunctions.m]: I was thinking about the `first: ` argument in the graphql query but it's not needed after all, I'll remove it

File: Artsy/Networking/conversations.graphql (hunk 3 above)
Comment (l2succes) [reply Artsy/Networking/conversations.graphql]: yes it could, I wanted to make the API forward compatible. I'll set it to 0 for now

**Comment by alloy** [comment]
Aight 👍

**Pull request merged**